libconfluo/confluo/container/cursor/json_cursors.h

```diff
@@ -0,0 +1,51 @@
+#ifndef CONFLUO_CONTAINER_CURSOR_JSON_CURSOR_H_
+#define CONFLUO_CONTAINER_CURSOR_JSON_CURSOR_H_
+
+#include <unordered_set>
+
+#include "batched_cursor.h"
+#include "offset_cursors.h"
+#include "schema/record.h"
+#include "parser/expression_compiler.h"
+#include "container/data_log.h"
+
+namespace confluo {
+
+typedef batched_cursor<std::string> json_cursor;
+
+/**
+ * A json cursor that make records into a json formatted string
+ */
+class aggregated_json_cursor : public json_cursor {
+ public:
+  /**
+   * Initializes the filter record
+   *
+   * @param o_cursor The offset cursor
+   * @param dlog The data log pointer
+   * @param schema The schema
+   * @param cexpr The filter expression
+   * @param batch_size The number of records in the batch
+   */
+  aggregated_json_cursor(std::unique_ptr<offset_cursor> o_cursor,
```

Comment:
This should take `std::unique_ptr<record_cursor>` as first argument, `const schema_t *`as second argument (for converting record to json) and `size_t batch_size` as the third and last argument.